GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1034,47 +1033,55 @@ void OnRevisionRead(GitRevision revision)
+                _workTreeChangeCount = new ArtificialCommitChangeCount();
 
-                    // Add working directory as virtual commit
-                    GitRevision workTreeRev = new(ObjectId.WorkTreeId)
-                    {
-                        Author = userName,
-                        AuthorUnixTime = 0,
-                        AuthorEmail = userEmail,
-                        Committer = userName,
-                        CommitUnixTime = 0,
-                        CommitterEmail = userEmail,
-                        Subject = ResourceManager.TranslatedStrings.Workspace,
-                        ParentIds = new[] { ObjectId.IndexId },
-                        HasNotes = true
-                    };
-                    _gridView.Add(workTreeRev);
-
-                    // Add index as virtual commit
-                    GitRevision indexRev = new(ObjectId.IndexId)
-                    {
-                        Author = userName,
-                        AuthorUnixTime = 0,
-                        AuthorEmail = userEmail,
-                        Committer = userName,
-                        CommitUnixTime = 0,
-                        CommitterEmail = userEmail,
-                        Subject = ResourceManager.TranslatedStrings.Index,
-                        ParentIds = new[] { filteredCurrentCheckout },
-                        HasNotes = true
-                    };
-                    _gridView.Add(indexRev);
-
-                    UpdateArtificialCommitCount(_artificialStatus, workTreeRev, indexRev);
-                }
+                var userName = Module.GetEffectiveSetting(SettingKeyString.UserName);
+                var userEmail = Module.GetEffectiveSetting(SettingKeyString.UserEmail);
+
+                // Add working directory as an artificial commit
+                GitRevision workTreeRev = new(ObjectId.WorkTreeId)
+                {
+                    Author = userName,
+                    AuthorUnixTime = 0,
+                    AuthorEmail = userEmail,
+                    Committer = userName,
+                    CommitUnixTime = 0,
+                    CommitterEmail = userEmail,
+                    Subject = ResourceManager.TranslatedStrings.Workspace,
+                    ParentIds = new[] { ObjectId.IndexId },
+                    HasNotes = true
+                };
+                _gridView.Add(workTreeRev, insertFirst: insertFirst);
+
+                // Add index as an artificial commit
+                GitRevision indexRev = new(ObjectId.IndexId)
+                {
+                    Author = userName,
+                    AuthorUnixTime = 0,
+                    AuthorEmail = userEmail,
+                    Committer = userName,
+                    CommitUnixTime = 0,
+                    CommitterEmail = userEmail,
+                    Subject = ResourceManager.TranslatedStrings.Index,
+                    ParentIds = new[] { CurrentCheckout },
```

Comment:
What is the difference between `CurrentCheckout` and - the formerly used - `filteredCurrentCheckout`?

Reply:
None, filteredCurrentCheckout was always set to CurrentCheckout.